config/types/files.go

```diff
@@ -63,9 +63,11 @@ type File struct {
 }
 
 type FileContents struct {
-	Remote Remote `yaml:"remote"`
-	Inline string `yaml:"inline"`
-	Local  string `yaml:"local"`
+	Base64      *bool  `yaml:"base64"`
+	Compression string `yaml:"compression"`
```

Comment:
I'm not a fan of having both `compression` and `remote.compression`. Supporting compression for local/inline files is good, but it should be consistent.

Reply:
Hi @ajeddeloh,

That's fine. I just did this to maintain backwards compatibility.

Reply:
Hi @ajeddeloh,

FWIW, I believe if compression is added to local content, it should be removed as a field from remote. As I said, I did it the way I did it to maintain back-compat.

Reply:
Hi @ajeddeloh,

Thoughts on this? I don't mind moving the `Compression` field up a level in the struct hierarchy. I just didn't know if y'all were okay with breaking backwards compatibility. Obviously `Compression` shouldn't exist as a field in `Remote` if the former applies to local content as well.

Reply:
There is no guarantee of backward compatibility with CT, so we can (and I think we should) move the field. The reason we don't need to worry about backward compatibility is because it's expected that users pin to a specific version of CT for their config. As they upgrade CT over time, they'll need to potentially update their configs. This would be one of those cases.

Reply:
Uh, we're not 100% sure that's the case. We failed to decide how to version ct configs and definitely didn't communicate anything along the lines of "hey, this spec is subject to change". I'm not sure what the best path forward is, but I'd be inclined to deprecrate the old field while adding the new one and removing after a set amount of time instead of just removing it with no warning.

Reply:
CT has not hit 1.0 and since the spec isn't versioned, it's also subject to the same versioning rules. We can use this opportunity to explain to users how to properly use the tool. I think this will be a good first test since there are probably very few people making use of compression.